h2o-core/src/main/java/water/rapids/ast/prims/advmath/AstStratifiedSplit.java

```diff
@@ -57,7 +57,7 @@ public static Vec split(Vec stratifyingColumn, double splittingFraction, long ra
     // Collect input vector domain
     final long[] classes = new VecUtils.CollectIntegerDomain().doAll(stratifyingColumn).domain();
     // Number of output classes
-    final int numClasses = stratifyingColumn.isNumeric() ? classes.length : stratifyingColumn.domain().length;
+    final int numClasses = classes.length;
```

Comment:
This is the actual fix. Domain of the stratifyingColumn (the Vec instance) remains the same, but there are less rows afterwards, means "high" probability of reducing the domain cardinality. This is the case here. CollectIntegerDomain always provides the up-to-date cardinality no matter what.